helios-client/src/main/java/com/spotify/helios/common/descriptors/ExecHealthCheck.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2014 Spotify AB.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.helios.common.descriptors;
+
+import com.google.common.base.Objects;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import org.jetbrains.annotations.Nullable;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+
+public class ExecHealthCheck extends HealthCheck {
+
+  private final String cmd;
+
+  public ExecHealthCheck(@JsonProperty("cmd") @Nullable final String cmd) {
+    super(TCP);
+    this.cmd = cmd;
+  }
+
+  private ExecHealthCheck(ExecHealthCheck.Builder builder) {
+    super(EXEC);
+    cmd = builder.cmd;
+  }
+
+  @Nullable
+  public String getCmd() {
+    return cmd;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+
+    ExecHealthCheck that = (ExecHealthCheck) o;
+
+    if (cmd != null ? !cmd.equals(that.cmd) : that.cmd != null) {
+      return false;
+    }
+
+    return true;
+  }
+
+  @Override
+  public int hashCode() {
+    int result = cmd != null ? cmd.hashCode() : 0;
+    return result;
+  }
+
+  @Override
+  public String toString() {
+    return Objects.toStringHelper(this)
+        .add("cmd", cmd)
+        .toString();
+  }
+
+  static Builder newBuilder() {
+    return new Builder();
+  }
+
+  public static class Builder {
```

Comment:
Several of our descriptors have `of` factory methods like [this](https://github.com/spotify/helios/blob/947d06c952422de0143455cfba57163e65debe3c/helios-client/src/main/java/com/spotify/helios/common/descriptors/PortMapping.java#L93). Might be nice to add `of` to these object as well since it's a little more concise than invoking the builder directly.